kong/clustering/services/sync/rpc.lua

```diff
@@ -0,0 +1,353 @@
+local _M = {}
+local _MT = { __index = _M, }
+
+
+local txn = require("resty.lmdb.transaction")
+local declarative = require("kong.db.declarative")
+local constants = require("kong.constants")
+local concurrency = require("kong.concurrency")
+
+
+local insert_entity_for_txn = declarative.insert_entity_for_txn
+local delete_entity_for_txn = declarative.delete_entity_for_txn
+local DECLARATIVE_HASH_KEY = constants.DECLARATIVE_HASH_KEY
+local CLUSTERING_SYNC_STATUS = constants.CLUSTERING_SYNC_STATUS
+local SYNC_MUTEX_OPTS = { name = "get_delta", timeout = 0, }
+
+
+local pairs = pairs
+local ipairs = ipairs
+local fmt = string.format
+local ngx_null = ngx.null
+local ngx_log = ngx.log
+local ngx_ERR = ngx.ERR
+local ngx_INFO = ngx.INFO
+local ngx_DEBUG = ngx.DEBUG
+
+
+-- number of versions behind before a full sync is forced
+local FULL_SYNC_THRESHOLD = 512
```

Comment:
Whats the reason for this specific number?

Reply:
KAG-5586